### PR TITLE
Fix bike check in delete

### PR DESCRIPTION
**Pull request by allenmichael099**
Switched to object ID to identify which Checkinout entry to delete. It used to use user_id, but that did not delete database entries because it was not converted to BSON before querying. I couldn't find a way to convert it to BSON or a way to query with BSON in mongolite.

### CODE REVIEW

File: R/mod_DTedit.R [hunk 1]
```diff
@@ -73,12 +73,14 @@ mod_DTedit_server <- function(input, output, session, table_data, table_type,
...
 
+  column_names <-  names(table_data)
+  
   return_values <- callModule(
     DTedit::dteditmod,
     id = "DTedit_table",
     thedata = table_data,
     edit.cols = edit_columns,
-    view.cols = names(table_data),
+    view.cols = column_names[!column_names %in% '_id'], # view the table without the '_id' column
```
Comment (asiripanich) [inline R/mod_DTedit.R:88]: '_id' is a column name right? not a suffix?

**Review by asiripanich — COMMENTED** [review]
Is this urgent? if not I would like to properly review it before we merge?

File: R/mod_DTedit.R (hunk 1 above)
Comment (allenmichael099) [reply R/mod_DTedit.R:88]: Right, '_id' is a column name here. I think it would be fine to wait until you've properly reviewed it.

**Comment by allenmichael099** [comment]
@shankari @asiripanich As far as I can tell, this is now ready to go. Checkinout objects get deleted based on objectId, and the formatted time still displays correctly.  

If in future edits to this feature you find the problem below, it is probably because the target indices within columnDefs got shifted out of place.
Error_no_bike_data

**Review by shankari — APPROVED** [review]
I think that the PR looks fine from an e-mission data model perspective, but I am not an R expert, and I don't know whether this is properly R-ic.

**Comment by asiripanich** [comment]
> this feature you find the problem below, it is probably because the target indices

I personally don't like this workaround. Maybe DTedit is a bad choice to begin with. However, I'm going to merge it to allow this feature to exist. But it will most likely be removed when I do a major update to the dashboard.

File: R/mod_DTedit.R [hunk 2]
```diff
@@ -28,12 +28,18 @@ mod_DTedit_server <- function(input, output, session, table_data, table_type,
...
+  # Target column indices start from 0 for columnDefs.
+  
+  # "user_id"   "_id"       "status"    "bikeLabel" "ts"        "fmt_time"
+  # fmt_time is column 6 in R, but without _id and with targets starting from 0,
+  # it should be changed to target 4
+  
   if ("columnDefs" %in% names(DT_options)) {
-    DT_options$columnDefs[[1]]$targets <- DT_options$columnDefs[[1]]$targets - 1
-  }
+    # Adjust for id and for fmt_time
```
Comment (asiripanich) [inline R/mod_DTedit.R:40]: I put `browser()` inside this but I never stopped here. When do we need this?

**Comment by shankari** [comment]
@asiripanich I don't know that the problem is with DTedit; the problem is with making the call to the database to delete the entry. We will have to solve something similar for the user status editing anyway.

**Review by asiripanich — COMMENTED** [review]
@allenmichael099

**Comment by asiripanich** [comment]
@allenmichael099 could you please explain why you need this? I know it is not part of this PR but seems related to this.

https://github.com/asiripanich/emdash/blob/e44b82c3be282d22e2ca0e11966116533c568028/R/app_server.R#L202-L217

File: R/mod_DTedit.R (hunk 2 above)
Comment (allenmichael099) [reply R/mod_DTedit.R:40]: This is for when we need to render a formatted time. Currently it only is needed for Bike Check In, which has a times stamp that we want to also display as a human readable time in a separate column. I'm surprised the browser() didn't stop - it stopped there for me just now.

**Comment by allenmichael099** [comment]
@asiripanich The code snippet above is so we will render a human readable time next to the timestamp every time a supplementary table has a column called 'ts'.

File: R/mod_DTedit.R (hunk 2 above)
Comment (asiripanich) [reply R/mod_DTedit.R:40]: Can you check that the test database you have is the same as the one in the repo?

**Comment by asiripanich** [comment]
> 
> 
> @asiripanich The code snippet above is so we will render a human readable time next to the timestamp every time a supplementary table has a column called 'ts'.

Can we convert this timestamp to a proper datetime field during the 'load data' step?

**Comment by shankari** [comment]
@asiripanich we tried using datetime, but since the conversion happens in R, it converts based on the timezone of the server, which for AWS is UTC. So instead of the user seeing the datetime in their local timezone, they will see it in UTC, and have to do the conversion in their head.

With this option, we convert the timestamp on the browser instead.

File: R/mod_DTedit.R (hunk 2 above)
Comment (allenmichael099) [reply R/mod_DTedit.R:40]: @asiripanich There probably was a difference, with mine containing a timestamp field and yours not. @shankari can provide you with data that has timestamps.

**Pull request merged**